drivers/sensors/sensor.c

```diff
@@ -651,22 +682,22 @@ int sensor_register(FAR struct sensor_lowerhalf_s *lower, int devno)
     {
       if (!lower->buffer_size)
         {
-          lower->buffer_size = g_sensor_info[lower->type].esize;
+          lower->buffer_size = esize;
+        }
+      else
+        {
+          lower->buffer_size = ROUNDUP(lower->buffer_size, esize);
         }
 
       lower->push_event = sensor_push_event;
     }
   else
     {
       lower->notify_event = sensor_notify_event;
+      lower->buffer_size = 0;
```

Comment:
I'm not following why do you need an extra function with an extra parameter when the lower half struct already gives this information directly. Isn't actually `lower->buffer_size` what is needed?

Reply:
The buffer size(buffer_size) is different from the element size(esize) since the sensor lower half normally generate the structural data stream. For examples, the custom sensor could define:
struct custem_event_s
{
 uint64_t timestamp;
  int16_t adc[8];
};
and call sensor_custom_register(lower, 1024, sizeof(struct custem_event_s), "/dev/sensor/custom0");
of course, the custom driver could pass 1 as esize if the generate the pure byte stream or variable size structural data.

Reply:
Ok. Seems strange to me to allow partial writes of a fixed struct. Would make more sense to me to specify the number of elements and the element size. But that is unrelated to this PR.

Reply:
After discussing, the buffer number is better than buffer size, we will provide a new PR to change the unit.

Reply:
Okay, I will modify this issue later.